monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -875,7 +875,29 @@ object Task extends TaskInstances {
     */
   def gather[A, M[X] <: TraversableOnce[X]](in: M[Task[A]])
     (implicit cbf: CanBuildFrom[M[Task[A]], A, M[A]]): Task[M[A]] =
-    TaskGather(in)(cbf)
+    TaskGather(in.toArray, () => cbf(in))
+
+   /** Given a `TraversableOnce[A]` and a function `A => Task[B]`,
+   * nondeterministically apply the function to each element of the collection
+   * and return a task that will signal a collection of the results once all
+   * tasks are finished.
+   *
+   * This function is the nondeterministic analogue of `traverse` and should
+   * behave identically to `traverse` so long as there is no interaction between
+   * the effects being gathered. However, unlike `traverse`, which decides on
+   * a total order of effects, the effects in a `wander` are unordered with
+   * respect to each other.
+   *
+   * Although the effects are unordered, we ensure the order of results
+   * matches the order of the input sequence. Also see [[wanderUnordered]]
+   * for the more efficient alternative.
+   *
+   * It's a generalized version of [[gather]].
+   */
+  def wander[A, B, M[X] <: TraversableOnce[X]](in: M[A])(f: A => Task[B])
+    (implicit cbf: CanBuildFrom[M[A], B, M[B]]): Task[M[B]] = {
+    TaskGather(in.map(f).toArray, () => cbf(in))
```

Comment:
I would have done it like ... 

```scala
def wander[A, B, M[X] <: TraversableOnce[X]](in: M[A])(f: A => Task[B])
  (implicit cbf: CanBuildFrom[M[A], A, M[A]]): Task[M[B]] = {

  Task.eval(in.map(f)).flatMap(col => TaskGather(col, () => cbf(in)))
}
```

This way `in.map(f)` is protected from thrown exceptions and you don't have to turn it into an array.